xwiki-platform-core/xwiki-platform-url/xwiki-platform-url-api/src/main/java/org/xwiki/url/URLSecurityManager.java

```diff
@@ -82,6 +91,10 @@ default boolean isURITrusted(URI uri)
      * This method throws a {@link SecurityException} if the parsed URI is not safe to use according to
      * {@link #isURITrusted(URI)}. It might also throw a {@link URISyntaxException} if the parameter cannot be properly
      * parsed.
+     * Note that this method might try to "repair" URI that are not parsed correctly by {@link URI#URI(String)}
+     * (e.g. serialized uri containing spaces). As part of that repair we need to perform escaping of the {@code %}
+     * character and we use {@link #PERCENT_ESCAPE} value as replacement string: the chain contained in
+     * {@link #PERCENT_ESCAPE} is then a protected token that should never be present in your serialized URI.
```

Comment:
Note that I even hesitated to add a check in the beginning of the implem to immediately return an error if the token is used. But it feels a bit too much.